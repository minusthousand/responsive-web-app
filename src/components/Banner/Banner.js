import React from 'react'

import BannerStyle from '../../layout/banner/index'
import Button from '../../ui/button/Button'

export function Banner() {
    const scrollToProducts = () => {
        window.scrollTo({
            top: 750,
            behavior: 'smooth',
        })
    }

    return (
        <BannerStyle>
            <BannerStyle.Video src="/videos/video.mp4" autoPlay loop muted />
            <BannerStyle.Title>
                Shoes you've been dreaming of.
            </BannerStyle.Title>
            <BannerStyle.Text>Check them right away!</BannerStyle.Text>
            <BannerStyle.Button>
                <Button
                    large
                    outlined
                    onClick={scrollToProducts}
                    className="btns"
                >
                    PRODUCTS
                    <BannerStyle.Icon className="fas fa-angle-down" />
                </Button>
            </BannerStyle.Button>
        </BannerStyle>
    )
}

export default Banner
