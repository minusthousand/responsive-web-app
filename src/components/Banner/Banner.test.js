import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Banner from './Banner'

// Necessary due to bug
// https://github.com/testing-library/react-testing-library/issues/470
Object.defineProperty(HTMLMediaElement.prototype, 'muted', {
    set: () => null,
})

Object.defineProperty(window, 'scrollTo', {
    configurable: true,
})
window.scrollTo = jest.fn()

describe('<Banner />', () => {
    it('should render', () => {
        const { container } = render(<Banner />)

        expect(container).toMatchSnapshot()
    })

    it('should call window scroll method when Products button was clicked', async () => {
        const { getByRole } = render(<Banner />)

        fireEvent.click(getByRole('button'))

        expect(window.scrollTo).toHaveBeenCalled()
    })
})
