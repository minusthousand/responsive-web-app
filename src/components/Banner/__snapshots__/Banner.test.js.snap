// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render 1`] = `
<div>
  <div
    class="banner-container"
  >
    <video
      autoplay=""
      loop=""
      src="/videos/video.mp4"
    />
    <h1>
      Shoes you've been dreaming of.
    </h1>
    <p>
      Check them right away!
    </p>
    <div
      class="banner-btns"
    >
      <div
        class="btn--mobile"
      >
        <button
          class="btn btn--large btn--outlined"
        >
          PRODUCTS
          <i
            class="fas fa-angle-down"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
